packages/yarnpkg-core/sources/Project.ts

```diff
@@ -470,6 +470,26 @@ export class Project {
     return null;
   }
 
+  async validateEverything(opts: {
+    validationWarnings: Array<{name: MessageName, text: string}>,
+    validationErrors: Array<{name: MessageName, text: string}>,
+    problemCount: number,
+    report: Report,
+  }) {
+    const progress = Report.progressViaCounter(opts.problemCount);
+    opts.report.reportProgress(progress);
```

Comment:
Is the progress useful? It seems something that will execute instantly

Reply:
Ah, right. The correct way to do this would be to trigger the hook inside `validateEverything`, but we don't know if we should call the `startTimer` function until we trigger the hook. 🤔

Reply:
Ignore my last comment. There's no way for the progress idea to make any sense. I'll just remove it.